index.html

```diff
@@ -1,11 +1,144 @@
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
     <meta charset="UTF-8">
-    <meta http-equiv="refresh" content="0;url=https://github.com/seriouslysean/monster-hunter-now-events">
-    <title>Redirecting...</title>
+    <title>Monster Hunter Events Today</title>
 </head>
-<body>
-    If you are not redirected, <a href="https://github.com/seriouslysean/monster-hunter-now-events">click here</a>.
+
+<body style="
+        background-color:#F6F4E4;
+        color: #222;
+        padding: 0px;
+        margin: auto;
+        display: flex;
+        flex-direction:column;
+        row-gap: 15px;
+        align-items: center;
+        text-align: center;
+        overflow-x: hidden;">
+    <!-- Page Heading -->
+    <div style="
+        font-size: 60px;
+        font-style: italic;
+        font-weight: 700;
+    ">Monster Hunter Events Today</div>
+
+    <!-- Date -->
+    <div style="
+        font-size: 40px;
+        font-style: italic;
+        font-weight: 700;" id="tag_display_date">
+
+    </div>
+
+    <!-- Display Event Cards Here -->
+    <div style="
+        display:flex;
+        flex-direction:column;
+        gap:20px;" id="tag_display_events">
+    </div>
+
+    <script type="text/javascript">
+
+        let eventsTodayList = [];
+        const TODAY = new Date();
+
+        function appendData(data) {
+            data.events.forEach((event) => {
+                // console.log("event: ", event);
+                let flag = 0;
```

Comment:
You likely don’t need a flag. You should be able to loop through the events and if it occurs today push that event directly in to the new collection.

I see what you were going for though, and the logic still works. The flag feels like unnecessary complexity.